coderd/database/databasefake/generator.go

```diff
@@ -0,0 +1,400 @@
+	if g.primaryOrg == nil {
+		org := g.Organization(ctx, database.Organization{
+			ID:          g.Lookup(primaryOrgName),
+			Name:        primaryOrgName,
+			Description: "This is the default primary organization for all tests",
+			CreatedAt:   time.Now(),
+			UpdatedAt:   time.Now(),
+		})
+		g.primaryOrg = &org
+	}
+
+	return *g.primaryOrg
+}
+
+func populate[DBType any](ctx context.Context, g *Generator, name string, seed DBType) DBType {
+	if name == "" {
+		name = g.RandomName()
+	}
+
+	out := g.Populate(ctx, map[string]interface{}{
+		name: seed,
+	})
+	v, ok := out[name].(DBType)
+	if !ok {
+		panic("developer error, type mismatch")
+	}
+	return v
+}
+
+func (g *Generator) RandomName() string {
+	for {
+		name := namesgenerator.GetRandomName(0)
+		if _, ok := g.names[name]; !ok {
+			return name
+		}
+	}
+}
+
+func (g *Generator) APIKey(ctx context.Context, seed database.APIKey) (key database.APIKey, token string) {
+	name := g.RandomName()
+	out := g.Populate(ctx, map[string]interface{}{
+		name: seed,
+	})
+	key, keyOk := out[name].(database.APIKey)
+	secret, secOk := out[name+"_secret"].(string)
+	require.True(g.testT, keyOk && secOk, "APIKey & secret must be populated with the right type")
+
+	return key, fmt.Sprintf("%s-%s", key.ID, secret)
+}
+
+func (g *Generator) File(ctx context.Context, seed database.File) database.File {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) UserLink(ctx context.Context, seed database.UserLink) database.UserLink {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) WorkspaceResource(ctx context.Context, seed database.WorkspaceResource) database.WorkspaceResource {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Job(ctx context.Context, seed database.ProvisionerJob) database.ProvisionerJob {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Group(ctx context.Context, seed database.Group) database.Group {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Organization(ctx context.Context, seed database.Organization) database.Organization {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Workspace(ctx context.Context, seed database.Workspace) database.Workspace {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Template(ctx context.Context, seed database.Template) database.Template {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) TemplateVersion(ctx context.Context, seed database.TemplateVersion) database.TemplateVersion {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) WorkspaceBuild(ctx context.Context, seed database.WorkspaceBuild) database.WorkspaceBuild {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) User(ctx context.Context, seed database.User) database.User {
+	return populate(ctx, g, "", seed)
+}
+
+func (g *Generator) Populate(ctx context.Context, seed map[string]interface{}) map[string]interface{} {
```

Comment:
Add a comment that this must be run in the main test goroutine as it uses `require` which calls `t.FailNow` which must be called from the goroutine running the test or benchmark function.